core/src/main/scala/kafka/tools/DumpLogSegments.scala

```diff
@@ -257,7 +257,7 @@ object DumpLogSegments {
     private def parseGroupMetadata(groupMetadataKey: GroupMetadataKey, payload: ByteBuffer) = {
       val groupId = groupMetadataKey.key
       val group = GroupMetadataManager.readGroupMessageValue(groupId, payload)
-      val protocolType = group.protocolType
+      val protocolType = group.protocolType.getOrElse("NONE")
```

Comment:
@ijuma Makes sense. I agree it would be awesome if there was stricter type checking for `==` by default. I can't think of any case where I wouldn't want that.